src/types.ts

```diff
@@ -93,7 +93,7 @@ export type NodeType =
 
 export type Primitive = null | boolean | number | string;
 
-export type RenderableTreeNode = Tag | string | null;
+export type RenderableTreeNode = Tag | Primitive;
```

Comment:
Technically, this is still not 100% accurate because `undefined` (or really anything) can be returned by the resolution of an `Ast.Variable` right?  Should we add `| object | undefined` here too then?

Reply:
It we want to include `object`, then we should use `Tag | Scalar`. However, if we use `Tag | Scalar | undefined`, feels like we might just want to use `any`?

Reply:
I am happy to use `any`, as I think that is the most correct option. There were some objections to that in slack, however.

Reply:
I think `any` is the most accurate.

Reply:
`any` doesn't give us any type safety and wouldn't have prevented the incident though right? https://www.typescriptlang.org/docs/handbook/basic-types.html#any

Reply:
I forgot this. Yeah let's do `Tag | Scalar` or `Tag | Scalar | undefined`

Reply:
If we do `| undefined`, we should probably just make `undefined` part of `Primitive`.